kratos/python_scripts/file_points_output_process.py

```diff
@@ -0,0 +1,118 @@
+            "file_path"         : "",
+            "sampling_points"   : 3,
+            "output_variables"  : [],
+            "historical_value"  : true,
+            "search_tolerance"  : 1e-6,
+            "print_format"      : "",
+            "output_file_settings": {}
+        }''')
+
+        params.ValidateAndAssignDefaults(default_settings)
+
+        # retrieving file path and name
+        file_path = params["file_path"].GetString()
+        if len(file_path) == 0:
+            raise Exception('A file path has has to be provided!')
+
+        # get and check number of sampling points
+        number_of_sampling_points = params["sampling_points"].GetInt()
+        if number_of_sampling_points <= 0:
+            raise Exception('The number of sampling points has to be larger than 0!')
+
+        # check the entity type
+        entity_type = params["entity_type"].GetString()
+        if not entity_type == "element" or entity_type == "condition":
+            raise Exception('"entity_type" can be either "element" or "condition"!')
+
+        # initialize position matrix for sampling points
+        positions = KratosMultiphysics.Matrix(number_of_sampling_points, 3)
+
+        # check and read file
+        if os.path.exists(file_path) and os.access(file_path, os.R_OK):
+
+            # open file
+            try:
+                with open(file_path, newline=None) as file:
+                    reader = csv.DictReader(file, delimiter=',')
+                    for p_count, row in enumerate(reader):
+
+                        # check if number of sampling points is already reached
+                        if p_count == number_of_sampling_points:
+                            KratosMultiphysics.Logger.PrintWarning("FilePointsOutputProcess: Number of points in file higher than number of samples!")
+                            break
+
+                        # add position of sampling point
+                        positions[p_count,0] = float(row['x'])
+                        positions[p_count,1] = float(row['y'])
+                        positions[p_count,2] = float(row['z'])
+
+            except:
+                raise Exception('The .csv-file can not be read (correctly)!')
+        else:
+            raise Exception('The file does not exist or can not be accessed!')
+
+        # check if enough points were found
+        p_count += 1
+        if p_count < number_of_sampling_points:
+            KratosMultiphysics.Logger.PrintWarning("FilePointsOutputProcess: Found " + str(p_count) + " points in file. Number of samples given was " + str(number_of_sampling_points) + "!")
+
+        params.RemoveValue("file_path")
+        params.RemoveValue("file_name")
+        params.RemoveValue("file_type")
+        params.RemoveValue("sampling_points")
+
+        params.AddEmptyValue("positions")
+        params["positions"].SetMatrix(positions)
+
+        self.multiple_points_output_process = MultiplePointsOutputProcess(model, params)
```

Comment:
Instead of "wraping" the process please return it from the `Factory` methods, see [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/postprocess_eigenvalues_process.py)
(I know that the other processes do that)

this has the bid advantage that if the interface of `Process` is extended the methods get automatically called

Reply:
I'm sorry, I don't understand. It looks to me like it is returned from Factory.

Reply:
instead of implementing your code inside a separate process, you should implement this inside the `Factory` method and return the instance of `MultiplePointsOutputProcess`